sway_libs/src/string/SPECIFICATION.md

```diff
@@ -0,0 +1,61 @@
+# Overview

```

Comment:
Here is a list of functions/traits I would like to see implemented in the String library that currently are missing.

`push_str`
`reserve`
`String::from(String | str)`

Reply:
`push_str` is not currently possible. A `str` of two different lengths are treated as seperate types and I would not be able to iterate over a `str`.

`reserve` should be implemented at the `vec` level, and then used here.

`String::from()` again has the same problem as `push_str`